### PR TITLE
Fix regression with directx when zfreeze=true and ztest=false.

### PR DESCRIPTION
**Pull request by phire**
Fixes [issue 8253](https://code.google.com/p/dolphin-emu/issues/detail?id=8253)

### CODE REVIEW

File: Source/Core/VideoCommon/PixelShaderGen.cpp [hunk 1]
```diff
@@ -275,8 +275,10 @@ static inline void GeneratePixelShader(T& out, DSTALPHA_MODE dstAlphaMode, API_T
 	                            && (g_ActiveConfig.bFastDepthCalc || bpmem.alpha_test.TestResult() == AlphaTest::UNDETERMINED)
 	                            // We can't allow early_ztest for zfreeze because depth is overridden per-pixel.
 	                            // This means it's impossible for zcomploc to be emulated on a zfrozen polygon.
-	                            && !bpmem.genMode.zfreeze;
-	const bool per_pixel_depth = (bpmem.ztex2.op != ZTEXTURE_DISABLE && bpmem.UseLateDepthTest()) || (!g_ActiveConfig.bFastDepthCalc && bpmem.zmode.testenable && !forced_early_z) || bpmem.genMode.zfreeze;
+	                            && !(bpmem.zmode.testenable && bpmem.genMode.zfreeze);
+	const bool per_pixel_depth = (bpmem.ztex2.op != ZTEXTURE_DISABLE && bpmem.UseLateDepthTest())
+	                             || (!g_ActiveConfig.bFastDepthCalc && bpmem.zmode.testenable && !forced_early_z)
+								 || (bpmem.zmode.testenable && bpmem.genMode.zfreeze);
```
Comment (magumagu) [inline Source/Core/VideoCommon/PixelShaderGen.cpp]: Incorrect tab usage.

File: Source/Core/VideoCommon/PixelShaderGen.cpp [hunk 2]
```diff
@@ -369,7 +371,7 @@ static inline void GeneratePixelShader(T& out, DSTALPHA_MODE dstAlphaMode, API_T
 		out.Write("void main(\n");
 		out.Write("  out float4 ocol0 : SV_Target0,%s%s\n  in float4 rawpos : SV_Position,\n",
 			dstAlphaMode == DSTALPHA_DUAL_SOURCE_BLEND ? "\n  out float4 ocol1 : SV_Target1," : "",
-			(per_pixel_depth && bpmem.zmode.testenable) ? "\n  out float depth : SV_Depth," : "");
+			(per_pixel_depth) ? "\n  out float depth : SV_Depth," : "");
```
Comment (magumagu) [inline Source/Core/VideoCommon/PixelShaderGen.cpp]: Extra parentheses.

File: Source/Core/VideoCommon/PixelShaderGen.cpp [hunk 3]
```diff
@@ -1030,7 +1032,7 @@ static inline void WriteAlphaTest(T& out, pixel_shader_uid_data* uid_data, API_T
 	uid_data->alpha_test_use_zcomploc_hack = bpmem.UseEarlyDepthTest()
 	                                         && bpmem.zmode.updateenable
 	                                         && !g_ActiveConfig.backend_info.bSupportsEarlyZ
-	                                         && !bpmem.genMode.zfreeze;
+											 && !bpmem.genMode.zfreeze;
```
Comment (magumagu) [inline Source/Core/VideoCommon/PixelShaderGen.cpp]: Tabs.

**Comment by JMC47** [comment]
confirmed  to fix the issue

**Pull request merged**